src/NServiceBus.AmazonSQS/TransportMessage.cs

```diff
@@ -29,15 +29,51 @@ public TransportMessage(OutgoingMessage outgoingMessage, List<DeliveryConstraint
             if (discardConstraint != null)
             {
                 Headers[TransportHeaders.TimeToBeReceived] = discardConstraint.MaxTime.ToString();
+                TimeToBeReceived = discardConstraint.MaxTime.ToString();
             }
 
             Body = outgoingMessage.Body != null ? Convert.ToBase64String(outgoingMessage.Body) : "empty message";
         }
 
-        public Dictionary<string, string> Headers { get; set; }
+        public Dictionary<string, string> Headers { get; set; } = new Dictionary<string, string>();
 
         public string Body { get; set; }
 
         public string S3BodyKey { get; set; }
+
+        public string TimeToBeReceived
+        {
+            get
+            {
+                if (timeToBeReceived == null)
+                {
+                    timeToBeReceived = Headers.ContainsKey(TransportHeaders.TimeToBeReceived) ? Headers[TransportHeaders.TimeToBeReceived] : TimeSpan.MaxValue.ToString();
```

Comment:
A backing field was used for `TimeToBeReceived`, but not for `ReplyToAddress`. Is it necessary?